packages/devextreme/js/__internal/core/speech_recognition_adapter.ts

```diff
@@ -0,0 +1,65 @@
+import { getWindow } from '@js/core/utils/window';
+import type { SpeechRecognitionConfig } from '@js/ui/speech_to_text';
+
+type SpeechRecognition = SpeechRecognitionConfig & {
+  start: () => void;
+  stop: () => void;
+} & Record<string, unknown>;
+
+interface SpeechRecognitionEvents {
+  onResult: (event: Event) => void;
+  onError: (event: Event) => void;
+  onEnd: () => void;
+}
+
+const SR_CONFIG_IGNORED_PROPERTIES = ['onresult', 'onerror', 'onend'];
```

Comment:
then there will be no custom abbreviation and it's clearer why it's ignored in config applying
```suggestion
const EVENT_NAMES = ['onresult', 'onerror', 'onend'];
```